objects/obj_settings_menu/Step_0.gml

```diff
@@ -0,0 +1,145 @@
+/// @description Control Settings Menu
+
+if (global.in_settings_menu)
+{
+	// move up
+	if ((keyboard_check_pressed(vk_up)) || keyboard_check_pressed(ord("W")))
+	{
+		if (global.settings_menu_cursor_position > 0)
+		{
+			global.settings_menu_cursor_position --;
+			// skip past empty spots
+			switch (global.settings_menu_cursor_position)
+			{
+				case 3:
+				case 6:
+					global.settings_menu_cursor_position --;
+					break;
+			}
+		}
+	}
+	
+	// move down
+	if ((keyboard_check_pressed(vk_down)) || keyboard_check_pressed(ord("S")))
+	{
+		if (global.settings_menu_cursor_position < (array_length(settings_menu_option) - 1))
+		{
+			global.settings_menu_cursor_position ++;
+			// skip past empty spots
+			switch (global.settings_menu_cursor_position)
+			{
+				case 3:
+				case 6:
+					global.settings_menu_cursor_position ++;
+					break;
+			}
+		}
+	}
+	
+	// switch for the settings menu options
+	switch (global.settings_menu_cursor_position)
+	{
+		// main volume
+		case 0:

```

Comment:
In general, for switch-case blocks, it would be good to create an ENUM that has more descriptive options than 0, 1, 2, and 7 (for example) for its cases. It allows for potential reusability in the codebase and for easier to read/understand code.

Reply:
If I'm understanding correctly, you want to add a layer between the settings_menu_cursor_position as an array index for the settings_menu_option array, to convert the numbers to words, then use those words in the case instead of the numbers.
Was the comment with each case number not clear enough, or just bad practice, or am I missing something?

Reply:
learned about and added ENUM for clarity